haystack/components/preprocessors/document_splitter.py

```diff
@@ -95,15 +97,39 @@ def _split_into_units(self, text: str, split_by: Literal["word", "sentence", "pa
             units[i] += split_at
         return units
 
-    def _concatenate_units(self, elements: List[str], split_length: int, split_overlap: int) -> List[str]:
+    def _concatenate_units(
+        self, elements: List[str], split_length: int, split_overlap: int
+    ) -> Tuple[List[str], List[int]]:
         """
         Concatenates the elements into parts of split_length units.
         """
         text_splits = []
+        splits_pages = []
+        cur_page = 1
         segments = windowed(elements, n=split_length, step=split_length - split_overlap)
         for seg in segments:
             current_units = [unit for unit in seg if unit is not None]
             txt = "".join(current_units)
             if len(txt) > 0:
                 text_splits.append(txt)
-        return text_splits
+                splits_pages.append(cur_page)
+                processed_units = current_units[: split_length - split_overlap]
+                if self.split_by != "page":
```

Comment:
Could you switch the logic here so that `if self.split_by == "page"` is the first statement? I think readability becomes easier, more intuitive.